src/Server/PrometheusRequestHandler.cpp

```diff
@@ -30,15 +30,8 @@ void PrometheusRequestHandler::handleRequest(HTTPServerRequest & request, HTTPSe
         response.setContentType("text/plain; version=0.0.4; charset=UTF-8");
 
         WriteBufferFromHTTPServerResponse wb(response, request.getMethod() == Poco::Net::HTTPRequest::HTTP_HEAD, keep_alive_timeout, write_event);
-        try
-        {
-            metrics_writer->write(wb);
-            wb.finalize();
```

Comment:
If `WriteBuffer::finalize()` throws an exception, it leave the write buffer in the "cancelled" state. It's not correct to call `finalize()` for a cancelled write buffer, see https://github.com/ClickHouse/ClickHouse/pull/64478/files#diff-58a49548f8d0a1c9a5ca3b19be09eee0cf283a27c654da666c02ef7d3e26f15dR124